addon/-private/properties/collection/main.js

```diff
@@ -0,0 +1,119 @@
+/* global Symbol */
+import Ember from 'ember';
+import { buildSelector, assign } from '../../helpers';
+import { create } from '../../create';
+import {
+  ELEMENT_NOT_FOUND,
+  throwBetterError
+} from '../../better-errors';
+import { count } from '../count';
+import Ceibo from 'ceibo';
+
+export class Collection {
+  constructor(definition, parent, key) {
+    this.definition = definition;
+    this.parent = parent;
+    this.key = key;
+
+    this._itemCounter = create({
+      count: count(definition.scope, {
+        resetScope: definition.resetScope,
+        testContainer: definition.testContainer
+      })
+    }, { parent });
+
+    this._items = [];
+  }
+
+  get length() {
+    return this._itemCounter.count;
+  }
+
+  objectAt(index) {
+    if (typeof this._items[index] === 'undefined') {
+      let { definition, parent, key } = this;
+      let scope = buildSelector({}, definition.scope, { at: index });
+
+      let finalizedDefinition = assign({}, definition);
+
+      finalizedDefinition.scope = scope;
+
+      let tree = create(finalizedDefinition, { parent });
+
+      // Change the key of the root node
+      Ceibo.meta(tree).key = `${key}[${index}]`;
+
+      this._items[index] = tree;
+    }
+
+    let itemPage = this._items[index];
+
+    if (index > this.length - 1) {
```

Comment:
What do you think about moving this up after line 35?
```javascript
throwBetterError(
  parent,
  `${key}[${index}]`,  // we could save this to a variable so as not to repeat it here and on line 44
  ELEMENT_NOT_FOUND,
  { selector: scope }
)
```

Reply:
Oh nah, I see why that won't work...would only work if an item hadn't yet been created for that index. But what do you think about changing the first two args as suggested above?